### ISSUE TITLE
Broken integration with script-executor plugin

### ISSUE THREAD

**Issue by tomzo** [gocd/gocd#2203]
##### Issue Type

- Bug Report

##### Summary

Started server with script executor plugin `0.2`. The following error is in the logs
```
java.lang.RuntimeException: Interaction with plugin with id 'script-executor' implementing 'task' extension failed while requesting for 'go.plugin-settings.get-configuration'. Reason: [null]
```
And plugin is not operational since start.
NullPointer suggests a bug IMHO.

###### Basic environment details

I ran this on master code + my changes, but I doubt that I broke it.

* Go Version: `16.5.0`
* JAVA Version: `7`
* OS: Linux

##### Log snippets

Complete message in log

Failed to fetch Plugin Settings metadata for plugin : script-executor java.lang.RuntimeException: Interaction with plugin with id 'script-executor' implementing 'task' extension failed while requesting for 'go.plugin-settings.get-configuration'. Reason: [null] at com.thoughtworks.go.plugin.access.PluginRequestHelper.submitRequest(PluginRequestHelper.java:38) at com.thoughtworks.go.plugin.access.common.settings.AbstractExtension.getPluginSettingsConfiguration(AbstractExtension.java:45) at com.thoughtworks.go.plugin.access.pluggabletask.TaskExtension.getPluginSettingsConfiguration(TaskExtension.java:69) at com.thoughtworks.go.plugin.access.common.settings.PluginSettingsMetadataLoader.fetchPluginSettingsMetaData(PluginSettingsMetadataLoader.java:68) at com.thoughtworks.go.plugin.access.common.settings.PluginSettingsMetadataLoader.pluginLoaded(PluginSettingsMetadataLoader.java:52) at com.thoughtworks.go.plugin.infra.DefaultPluginManager$FilterChangeListener.pluginLoaded(DefaultPluginManager.java:262) at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework$2.execute(FelixGoPluginOSGiFramework.java:353) at org.apache.commons.collections.CollectionUtils.forAllDo(CollectionUtils.java:389) at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework.getBundle(FelixGoPluginOSGiFramework.java:111) at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework.loadPlugin(FelixGoPluginOSGiFramework.java:97) at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.refreshBundle(DefaultPluginJarChangeListener.java:152) at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.addPlugin(DefaultPluginJarChangeListener.java:109) at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.pluginJarAdded(DefaultPluginJarChangeListener.java:72) at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners$1.execute(DefaultPluginJarLocationMonitor.java:288) at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners.doOnAllPluginJarChangeListener(DefaultPluginJarLocationMonitor.java:319) at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners.pluginJarAdded(DefaultPluginJarLocationMonitor.java:286) at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.notifyListenersOfAddedPlugins(DefaultPluginJarLocationMonitor.java:210) at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.loadAndNotifyPluginsFrom(DefaultPluginJarLocationMonitor.java:198) at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.run(DefaultPluginJarLocationMonitor.java:184) Caused by: java.lang.NullPointerException at com.thoughtworks.go.plugin.access.PluginRequestHelper.submitRequest(PluginRequestHelper.java:33) ... 18 more

##### Any other info

I recall some recent gitter conversation regarding changes around plugin view, but I wasn't involved. Not sure if this needs core update or plugin.

**Comment by tomzo** [comment]
Just confirmed it is on master. The reason is that server sends api request `go.plugin-settings.get-configuration` to plugin and response from plugin is null
```
GoPluginApiResponse response = pluginManager.submitTo(pluginId, apiRequest);
            if (DefaultGoApiResponse.SUCCESS_RESPONSE_CODE == response.responseCode()) {
```

**Comment by tomzo** [comment]
If I understand correctly, now these messages have changed names:
 * `configuration` to `go.plugin-settings.get-configuration`
 * `validate` to `go.plugin-settings.validate-configuration`
 * `view` to `go.plugin-settings.get-view`

I submitted a PR in plugin to support both new and old names.
In server I submitted a PR to handle null response with nicer message.

**Comment by ketan** [comment]
When did the request names change?

**Comment by tomzo** [comment]
I just found this #2098 and #1355. I can confirm that task plugin receives these requests
```
go.plugin-settings.get-configuration
go.plugin-settings.get-view
go.plugin-settings.validate-configuration
configuration
view
validate
execute
```
script-executor plugin did not handle the `go.plugin-settings-*` requests so far and it was not causing any problems in `16.2.0`. Recently I upgraded to `16.5.0` and not handling these messages causes plugin to be useless.
I think #1355 is the right way to solve this, make all requests namespaced. But that is quite some work. What I did here is minimal to get the plugin working again.
